### ISSUE TITLE
Improve test report details

### ISSUE THREAD

**Issue by jasonmp85** [citusdata/the-process#9]
It sure would be nice to have things like error logs, differences in test output, etc., right in our UI. I think CircleCI should be able to do this. Their [documentation on Collecting Test Metadata](https://circleci.com/docs/2.0/collect-test-data/) gives a high-level overview, but their reliance on the JUnit format (which is documented only in a few ancient XSD files floating around the web) means we'll need to use other implementations as inspiration.

Fortunately, CircleCI themselves ~~forked~~ _assumed maintainership_ of `minitest`, so we can look at precisely how they want their test output generated here: https://github.com/circleci/minitest-ci/blob/8b72b0f32f154d91b53d63d1d0a8a8fb3b01d726/lib/minitest/ci_plugin.rb#L131-L161

Another good reference is the RSpec JUnit formatter: https://github.com/sj26/rspec_junit_formatter/blob/8f981d20979a9efa235cc878790b92d8862ba560/lib/rspec_junit_formatter.rb#L14-L182

Basically, look at what they do and do something similar for the properties we care about in our tests.
